### PR TITLE
feat: add contract interaction transaction type

### PR DESCRIPTION
**Pull request by ragozin-nikita**
Fixes #

## Proposed Changes

  - HQM-106
  -
  -

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/transaction-detail.tsx [hunk 1]
```diff
@@ -77,42 +89,56 @@ export const TransactionDetail = ({
...
-        t14
-        style={styles.amount}
-      />
-      <Text
-        t6
-        color={isSent ? Color.textRed1 : Color.textGreen1}
-        style={styles.sum}>
-        {total} ISLM
-      </Text>
+      {!isContract && (
```
Comment (vivalaakam) [inline src/components/transaction-detail.tsx:92]: !total.sum

File: src/components/transaction-detail.tsx [hunk 2]
```diff
@@ -77,42 +89,56 @@ export const TransactionDetail = ({
...
+                : I18N.transactionDetailReciveFrom
+            }
+            reversed
+            short
+            onPress={onPressAddress}
+          />
+        )}
+        {isContract && contractName && (
+          <DataContent
+            subtitle={'Contract name'}
```
Comment (vivalaakam) [inline src/components/transaction-detail.tsx:136]: i18n pls =)

File: src/components/transaction-detail.tsx [hunk 3]
```diff
@@ -77,42 +89,56 @@ export const TransactionDetail = ({
...
+          <Text
+            i18n={I18N.transactionDetailTotalAmount}
+            t14
+            style={styles.amount}
+          />
+          <Text
+            t6
+            color={isSent ? Color.textRed1 : Color.textGreen1}
+            style={styles.sum}>
+            {total} ISLM
```
Comment (vivalaakam) [inline src/components/transaction-detail.tsx:103]: i18n pls

File: src/components/transactions/contract.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+export const TransactionContract = ({
+  item,
+  onPress,
+}: TransactionPreviewProps) => {
+  const [contractName, setContractName] = useState('');
+
+  useEffectAsync(async () => {
+    const name = await Indexer.instance.getContractName(item.to);
```
Comment (vivalaakam) [inline src/components/transactions/contract.tsx:25]: move it on screen layer

Comment (ragozin-nikita) [reply src/components/transactions/contract.tsx:25]: We should fetch contract name only when we display contract list item. On the screen layer we have renderItem method only

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**